exercise/static/exercise/chapter.js

```diff
@@ -579,31 +579,100 @@
           });
         });
       } else {
-        var exercise = this;
-        if (exercise.quiz) {
-          exercise.quizSubmission("submit");
-        } else {
-          /* TODO: migrate to overlay */
-          chapter.openModal(chapter.messages.submit);
+        // Callback for submitting an exercise
+        var submitCallback = function(exercise, hash) {
+          if (exercise.quiz) {
+            exercise.quizSubmission("submit");
+          } else {
+            /* TODO: migrate to overlay */
+            chapter.openModal(chapter.messages.submit);
+          }
+          var url = $(form_element).attr("action");
+          var formData = new FormData(form_element);
+          formData.append("__submission_hash", hash);
+
+          exercise.submitAjax(url, formData, function(data) {
+            //$(form_element).find(":input").prop("disabled", false);
+            //exercise.hideLoader();
+            var input = $(data);
+            if (exercise.quiz) {
+              exercise.quizSubmission("success");
+              exercise.update(input);
+              exercise.renderMath();
+              exercise.dom_element.dispatchEvent(
+                new CustomEvent("aplus:submission-finished",
+                  {bubbles: true, detail: {type: exercise.exercise_type}}));
+            } else {
+              exercise.updateSubmission(input);
+            }
+          });
+        };
+
+        let isDuplicate = false;
+        // Retrieve all the hashes of the previous submissions from the HTML
+        const hashes = new Set();
+        const datahashElements = this.element.find("[data-hash]");
+        for (let elem of datahashElements) {
+          try {
+            hashes.add(elem.dataset["hash"]);
```

Comment:
If the hashes are stored in a list/array and the indexes match the order of the submissions, then you would later know which submission is the duplicate one. It would be best to show the duplicate submission number to the user so that they can more easily check the submission contents.

Reply:
Yes, I had this idea too, but the problem I was facing was that I wasn't sure how I can pass the variable containing the duplicate submission number to the HTML modal template's `blocktranslate`. How can this be done?

Reply:
I answered in the review today:
https://github.com/apluslms/a-plus/pull/1048#pullrequestreview-1058583782

Basically, you can use client-side JS to update the duplicate submission number in the modal contents. The number is not known at the time when the html template is rendered in the Django backend.